lib/core/presentation/pages/home/home_page.dart

```diff
@@ -64,10 +67,22 @@ class _HomeListingView extends StatefulWidget {
 }
 
 class _HomePageViewState extends State<_HomeListingView> {
+  final _appRouter = AppRouter();
+
   @override
   void initState() {
     super.initState();
     getIt<ShakeService>().startShakeDetection(context);
+
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      getIt<MatrixService>().backgroundPush.setupContextAndRouter(
+            router: _appRouter,
```

Comment:
please move this to _App widget.
Homescreen should not contain this logic

Reply:
Resolved